gotocompany/optimus/core/v1beta1/resource.proto

```diff
@@ -140,6 +148,26 @@ message UpdateResourceResponse {
   string message = 2;
 }
 
+message UpsertResourceRequest {
+  string project_name = 1;
+  string datastore_name = 2;
+  repeated ResourceSpecification resources = 3;
+  string namespace_name = 4;
+}
+
+message UpsertResourceResponse {
+  bool success = 1;
+  string message = 2;
+  repeated ResourceStatus results = 3;
+  repeated string successful_resource_names = 4;
```

Comment:
@sbchaos even though we do have `results`, `successful_resource_names` is required for registering only the successful resources through Shield proxy. let me know if you have any concern on this.

Reply:
Should be fine.